src/validation/rules/file_name.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright 2023 Helsing GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::*;
+use std::path::PathBuf;
+
+/// Ensure that file names match package names.
+///
+/// For example, a package named `physics` should be stored in `physics.proto`. A package named
+/// `physics.rotation` should be stored in `physics/rotation.proto`.
+#[derive(Debug, Clone, Default)]
+pub struct FileName;
+
+fn file_name(package_name: &str) -> PathBuf {
+    format!("{}.proto", package_name.replace('.', "/")).into()
+}
+
+impl Rule for FileName {
+    fn rule_info(&self) -> &'static str {
+        "making sure file names matches package names"
+    }
+
+    fn check_package(&mut self, package: &Package) -> Violations {
+        let candidate = file_name(&package.name);
+        let correct = candidate == package.file;
+        if correct {
+            Violations::default()
+        } else {
+            let message = violation::Message {
+                message: format!(
+                    "file name should be {candidate:?} but is {:?}",
+                    package.file
+                ),
+                help: "Try to rename the file to align with the package name.".into(),
+            };
+            vec![self.to_violation(message)]
+        }
+    }
```

Comment:
Background: Using early returns instead of "if-else" helps to unnest code and make it way simpler to read as you have only to think about "exit conditions" instead of following the branched logic in your head

Reply:
Please resolve once implemented, otherwise it is hard to track